### PR TITLE
Fix CICS Formattime Command

### PR DESCRIPTION
**Pull request by KutluOzel-b**
## Checklist:
- [x] Each of my commits contains one meaningful change
- [x] I have performed rebase of my branch on top of the development
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSFormatTimeOptionsCheckUtility.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+                    put(CICSLexer.TIME, ErrorSeverity.WARNING);
+                    put(CICSLexer.TIMESEP, ErrorSeverity.WARNING);
+                    put(CICSLexer.YEAR, ErrorSeverity.WARNING);
+                    put(CICSLexer.YYDDD, ErrorSeverity.WARNING);
+                    put(CICSLexer.YYDDMM, ErrorSeverity.WARNING);
+                    put(CICSLexer.YYMMDD, ErrorSeverity.WARNING);
+                    put(CICSLexer.YYYYDDD, ErrorSeverity.WARNING);
+                    put(CICSLexer.YYYYDDMM, ErrorSeverity.WARNING);
+                    put(CICSLexer.YYYYMMDD, ErrorSeverity.WARNING);
+                }
```
Comment (ilidio-lopes) [inline server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSFormatTimeOptionsCheckUtility.java:68]: Error for parametrized options

**Review by ilidio-lopes — CHANGES_REQUESTED** [review]

**Comment by ilidio-lopes** [comment]
Has conflicts

File: server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSFormatTimeOptionsCheckUtility.java (hunk 1 above)
Comment (KutluOzel-b) [reply server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSFormatTimeOptionsCheckUtility.java:68]: adressed in latest commit

**Review by ilidio-lopes — APPROVED** [review]

File: server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/TestCICSFormatTimeStatement.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+            + "YYMMDD({$varFour}) YYYYDDD({$varFour}) \n"
+            + "YYYYDDMM({$varFour}) YYYYMMDD({$varFour})";
+    private static final String FORMAT_TIME_VALID =
+            "FORMATTIME ABSTIME({$varFour})";
+    private static final String FORMAT_TIME_MISSING_ABSTIME_INVALID =
+            "FORMATTIME {DATE(100) | errorMissingAbstime}";
+    private static final String FORMAT_TIME_MISSING_DATESTRING_INVALID =
+            "FORMATTIME {ABSTIME(100) STRINGZONE(100) | errorMissingDatestring}";
+    private static final String FORMAT_TIME_MISSING_TIME_INVALID =
+            "FORMATTIME {ABSTIME(100) TIMESEP | errorMissingTime}";
```
Comment (slavek-kucera) [inline server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/TestCICSFormatTimeStatement.java]:
```
DFH7226I           E        00002         SUB-OPERAND '100' OF 'DATE' KEYWORD SHOULD BE A DATA REFERENCE NOT AN   
                                          EXPRESSION OR A CONSTANT. COMMAND NOT TRANSLATED.                       
DFH7226I           E        00005         SUB-OPERAND '100' OF 'ABSTIME' KEYWORD SHOULD BE A DATA REFERENCE NOT AN
                                          EXPRESSION OR A CONSTANT. COMMAND NOT TRANSLATED.                       
DFH7226I           E        00008         SUB-OPERAND '100' OF 'ABSTIME' KEYWORD SHOULD BE A DATA REFERENCE NOT AN
                                          EXPRESSION OR A CONSTANT. COMMAND NOT TRANSLATED.                       
```

**Review by slavek-kucera — APPROVED** [review]

File: server/engine/src/main/antlr4/org/eclipse/lsp/cobol/implicitDialects/cics/CICSLexer.g4 [hunk 3]
```diff
@@ -793,6 +793,7 @@ STATUSTEXT : S T A T U S T E X T;
 STORAGE : S T O R A G E;
 STRFIELD : S T R F I E L D;
 STRINGFORMAT : S T R I N G F O R M A T;
+STRINGZONE: S T R I N G Z O N E;
```
Comment (slavek-kucera) [inline server/engine/src/main/antlr4/org/eclipse/lsp/cobol/implicitDialects/cics/CICSLexer.g4:797]: Should this be also added to the `cicsLexerDefinedVariableUsageTokens` rule?

**Pull request merged**